docs/thirdparty-openapi3-snippets.yaml

```diff
@@ -440,57 +477,87 @@ components:
         - scopes
         - participantId
         - initiatorId
-        - credential
-    ConsentGenerateChallengeIDPostRequest:
-      title: ConsentGenerateChallengeIDPostRequest
-      type: object
+    ConsentRequestChannelType:
+      title: ConsentRequestChannelType
+      type: string
+      enum:
+        - WEB
+        - OTP
       description: |
-        A credential used to allow a user to prove their identity
-        and access to an account with a DFSP.
+        The auth channel being used for the consentRequest.
+        - "WEB" - The Web auth channel.
+        - "OTP" - The OTP auth channel.
+    ConsentRequestIDPutResponse:
+      title: ConsentRequestIDPutResponse
+      type: object
+      description: 'The object sent in a `PUT /consentRequests/{ID}` request.'
       properties:
-        type:
-          $ref: '#/components/schemas/CredentialType'
+        id:
+          $ref: '#/components/schemas/ConsentRequestPostRequest/properties/id'
+        initiatorId:
+          type: string
+          description: The id of the PISP who will initiate transactions on a user's behalf.
+        scopes:
+          type: array
+          items:
+            $ref: '#/components/schemas/Scope'
+        authChannels:
+          type: array
+          items:
+            $ref: '#/components/schemas/AuthorizationChannelType'
+        callbackUri:
+          type: string
+          description: The callback uri that the user will be redirected to after completing the WEB auth channel.
+        authUri:
+          type: string
+          description: The callback uri that the pisp app redirects to for user to complete their login.
+        authToken:
+          type: string
+          description: The Auth token from the OTP or redirect to pisp app.
       required:
-        - type
-    ConsentIDPutRequest:
-      title: ConsentIDPutRequest
+        - id
+        - initiatorId
+        - scopes
+        - authChannels
+        - callbackUri
+        - authUri
+        - authToken
+    ConsentRequestPostRequest:
+      title: ConsentRequestPostRequest
       type: object
-      description: Data model for the complex type Consent.
+      description: The object sent in a `POST /consentRequests` request.
       properties:
-        requestId:
-          allOf:
-            - $ref: '#/components/schemas/AuthorizationPostRequest/properties/transactionId'
+        id:
```

Comment:
If I remember correctly the Consent.id is a key-handler which identified pair of public keys stored in User's device - are we sure is this id really UUID/CorrelationId format? What do you think @lewisdaly && @jgeewax

Reply:
@eoln  I think you're referring to `consent.credential.id` ConsentId is determined by the DFSP.

https://camo.githubusercontent.com/42b253dfcf3a8223cd6b056362872fdb9a5b9396/687474703a2f2f7777772e706c616e74756d6c2e636f6d2f706c616e74756d6c2f70726f78793f63616368653d6e6f267372633d68747470733a2f2f7261772e67697468756275736572636f6e74656e742e636f6d2f6d6f6a616c6f6f702f706973702f6d61737465722f646f63732f6c696e6b696e672f342d6772616e742d636f6e73656e742e70756d6c

Reply:
thanks @lewisdaly for enlightment